File: features/notes/src/main/java/com/nivasafinance/features/notes/exception/NotesOperationException.java
package com.nivasafinance.features.notes.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

public class NotesOperationException extends RuntimeException {
    private static final long serialVersionUID = 1L;

    public NotesOperationException(String operation, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(
                operation,
                new Object[]{},
                messageSource
        ));
    }
}

File: features/notes/src/main/java/com/nivasafinance/features/notes/exception/NotesOperationException.kt
package com.nivasafinance.features.notes.exception

import com.nivasafinance.common.exception.ExceptionUtils
import org.springframework.context.MessageSource

class NotesOperationException(
    operation: String,
    messageSource: MessageSource
) : RuntimeException(
    ExceptionUtils.createLocalizedMessage(
        operation,
        emptyArray(),
        messageSource
    )
)
